### PR TITLE
Added TeamCymruScout Solution

### PR DESCRIPTION
**Pull request by niralishah-crest**
Required items, please complete
   
   Change(s):
   - Added Team Cymru Scout Solution

   Reason for Change(s):
   - New Product

   Version Updated:
   - Initial Version 3.0.0

   Testing Completed:
   - Yes

   Checked that the validations are passing and have addressed any issues that are present:
   - Yes

### CODE REVIEW

**Comment by v-prasadboke** [comment]
Hello @niralishah-crest, Please try to resolve KQL and Logo validation errors

**Comment by niralishah-crest** [comment]
@v-prasadboke, we are trying to fix it from our end and will let you know as soon as possible.

**Comment by niralishah-crest** [comment]
@v-prasadboke We have fixed Logo validation error but still we are unable to find the reason behind KQLValidation error. Can you please help us out on that?
Thanks.

File: Solutions/Team Cymru Scout/Data Connectors/TeamCymruScout/SharedCode/checkpoint_manager.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    def __init__(self, file_path) -> None:
+        """
+        Initialize the insatnce object of CheckpointManager.
+
+        Args:
+            file_path (str): path of the file.
+        """
+        self.state = StateManager(
+            connection_string=consts.CONN_STRING, file_path=file_path
+        )
```
[the remark this answers is not in the document]
Comment (niralishah-crest) [reply Solutions/Team Cymru Scout/Data Connectors/TeamCymruScout/SharedCode/checkpoint_manager.py:22]: @v-prasadboke Here, the error reference is from NetskopeDataConnector and not from TeamCymruScout
image

File: Solutions/Team Cymru Scout/Data Connectors/TeamCymruScout/SharedCode/utils.py [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+        Returns:
+            list or None: A list of values from the watchlist if the indicator type is found,
+                          otherwise None.
+
+        Raises:
+            TeamCymruScoutException: If an error occurs while retrieving the data.
+        """
+        __method_name = inspect.currentframe().f_code.co_name
+        try:
+            logs_data, flag = get_logs_data(self.query_constants.get(indicator_type))
```
[the remark this answers is not in the document]
Comment (niralishah-crest) [reply Solutions/Team Cymru Scout/Data Connectors/TeamCymruScout/SharedCode/utils.py:166]: @v-prasadboke Same for here as well. Error reference is from BitSightDataConnector and not from TeamCymruScout.
image

File: Solutions/Team Cymru Scout/Data Connectors/TeamCymruScout/SharedCode/utils.py [hunk 3]
```diff
@@ -0,0 +1,246 @@
...
+        for label, params in required_params.items():
+            if not params or params == "":
+                missing_required_field = True
+                applogger.error(
+                    '{}(method={}) : "{}" field is not set in the environment please set '
+                    "the environment variable and run the app.".format(
+                        self.logs_starts_with,
+                        __method_name,
+                        label,
+                    )
```
[the remark this answers is not in the document]
Comment (niralishah-crest) [reply Solutions/Team Cymru Scout/Data Connectors/TeamCymruScout/SharedCode/utils.py:72]: @v-prasadboke Here, one of the error indicates the issue in environment variable name AZURE_CLIENT_SECRET as seen in attached screenshot. As mentioned [here](https://learn.microsoft.com/en-us/python/api/overview/azure/identity-readme?view=azure-python#service-principal-with-secret) we need to keep EnvironmentVariables same. 

The other errors are from CofenceTriage, CofenseIntelligence and BitSight Data Connector.

**Comment by v-prasadboke** [comment]
Hello @niralishah-crest, Can you please try to resolve branch Conflicts

**Comment by niralishah-crest** [comment]
> Hello @niralishah-crest, Can you please try to resolve branch Conflicts

@v-prasadboke I have resolved the branch conflicts.

**Comment by niralishah-crest** [comment]
@v-prasadboke Any updates on this PR review?

**Comment by niralishah-crest** [comment]
@v-prasadboke Current CodeQL failures are from other integrations as I mentioned in above comments. Can you please look into it?
Thanks.

**Comment by jayeshprajapaticrest** [comment]
@v-prasadboke 
Any update on the above PR?
Can you please proceed to review and get it merge as early as possible as this needs to be live as early as possible?
Its been around 1 and half month.

**Comment by v-prasadboke** [comment]
Hello @niralishah-crest & @jayeshprajapaticrest, Can you please share working images of Function app. Also please update this branch from master once

**Comment by niralishah-crest** [comment]
@v-prasadboke Please find the attached screenshot for running function app.
image
image

**Comment by jayeshprajapaticrest** [comment]
@v-prasadboke 
As mentioned above we have attached working screenshots of the Azure Function App(data connector). Can you please proceed with the review and get it merge ASAP?
Thanks in advanced.

**Comment by niralishah-crest** [comment]
> There are still some Code QL error failing

@v-prasadboke I have made the changes in code. Can you plese re-run the CodeQl?

**Comment by v-prasadboke** [comment]
Hello @niralishah-crest and @jayeshprajapaticrest there are still some Code-QL failure for secrets 
Please use another keyword

**Comment by niralishah-crest** [comment]
> Hello @niralishah-crest and @jayeshprajapaticrest there are still some Code-QL failure for secrets Please use another keyword

@v-prasadboke As mentioned in above [reply](https://github.com/Azure/Azure-Sentinel/pull/10939#discussion_r1741827136) we need to use AZURE_CLIENT_SECRET only as Environment variable and retrieve its value in code.

**Review by v-prasadboke — APPROVED** [review]

**Review by v-atulyadav — APPROVED** [review]

**Pull request merged**